### PR TITLE
Some bug fixes for VerticallyStretchedGrids

### PR DESCRIPTION
**Pull request by tomchor**

### CODE REVIEW

**Review by glwagner — APPROVED** [review]
Should we add a test too? Might just require running one of the existing NetCDFOutputWriter tests for VerticallyStretchedRectilinearGrid?

File: src/OutputWriters/netcdf_output_writer.jl [hunk 1]
```diff
@@ -36,12 +36,12 @@ function default_dimensions(output, grid, field_slicer)
...
-        "yF" => grid.yF.parent[parent_slice_indices(Face, TY, Ny, Hy, field_slicer.j, field_slicer.with_halos)],
-        "zC" => grid.zC.parent[parent_slice_indices(Center, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)],
-        "zF" => grid.zF.parent[parent_slice_indices(Face, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)])
+        "xC" => all_x_nodes(Center, grid)[parent_slice_indices(Center, TX, Nx, Hx, field_slicer.i, field_slicer.with_halos)],
+        "xF" => all_x_nodes(Face, grid)[parent_slice_indices(Face, TX, Nx, Hx, field_slicer.i, field_slicer.with_halos)],
+        "yC" => all_y_nodes(Center, grid)[parent_slice_indices(Center, TY, Ny, Hy, field_slicer.j, field_slicer.with_halos)],
+        "yF" => all_y_nodes(Face, grid)[parent_slice_indices(Face, TY, Ny, Hy, field_slicer.j, field_slicer.with_halos)],
+        "zC" => all_z_nodes(Center, grid)[parent_slice_indices(Center, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)],
+        "zF" => all_z_nodes(Face, grid)[parent_slice_indices(Face, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)])
```
Comment (ali-ramadhan) [inline src/OutputWriters/netcdf_output_writer.jl:44]: Kinda ugly but might fix tests?

```suggestion
        "xC" => all_x_nodes(Center, grid).parent[parent_slice_indices(Center, TX, Nx, Hx, field_slicer.i, field_slicer.with_halos)],
        "xF" => all_x_nodes(Face, grid).parent[parent_slice_indices(Face, TX, Nx, Hx, field_slicer.i, field_slicer.with_halos)],
        "yC" => all_y_nodes(Center, grid).parent[parent_slice_indices(Center, TY, Ny, Hy, field_slicer.j, field_slicer.with_halos)],
        "yF" => all_y_nodes(Face, grid).parent[parent_slice_indices(Face, TY, Ny, Hy, field_slicer.j, field_slicer.with_halos)],
        "zC" => all_z_nodes(Center, grid).parent[parent_slice_indices(Center, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)],
        "zF" => all_z_nodes(Face, grid).parent[parent_slice_indices(Face, TZ, Nz, Hz, field_slicer.k, field_slicer.with_halos)])
```

**Review by ali-ramadhan — APPROVED** [review]
Happy to merge if this potentially fixes NetCDF output for stretched grids.

Definitely agree a test would be good but I'm working on a PR that tests output for different grids in more detail so I could cover it there. Might also need more refactoring to support the lat-lon grid (and especially the cubed sphere).

I say this but I can't find the branch locally and I might have forgotten to to push it =/

**Comment by tomchor** [comment]
> Definitely agree a test would be good but I'm working on a PR that tests output for different grids in more detail so I could cover it there. 

That's okay by me. All I did was fix a bug so, even if we don't test this right now, the change is still positive. And you could do it in time for 0.53.2. Let's wait to see if your change fixes the tests!

**Pull request merged**